File: ios/MappyMapViewManager.m
//
//  MappyMapView.m
//  SnowMappyRN
//
//  Created by Afsar Shakeer on 2023-08-03.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(MappyMapViewManager, RCTViewManager)

RCT_EXTERN_METHOD(
  setMap:(nonnull NSNumber *)node
  venueInfo:(nonnull NSDictionary *)info
)

RCT_EXTERN_METHOD(
  rotateToNorth:(nonnull NSNumber *)node
)

RCT_EXTERN_METHOD(
  setInitialViewpoint:(nonnull NSNumber *)node
)

@end

File: ios/MappyMapViewManager.swift
//
//  MappyMapViewManager.swift
//  SnowMappyRN
//
//  Created by Afsar Shakeer on 2023-08-03.
//

import Foundation

@objc(MappyMapViewManager)
final class MappyMapViewManager: RCTViewManager {
  override func view() -> UIView! {
    return MappyMapViewContainer()
  }
  
  @objc
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func setMap(_ node: NSNumber, venueInfo: NSDictionary) {
      DispatchQueue.main.async {
        self.mappyMapViewContainer(node).setMap(venueInfo)
      }
  }
  
  @objc
  func rotateToNorth(_ node: NSNumber) {
    DispatchQueue.main.async {
      self.mappyMapViewContainer(node).rotateToNorth()
    }
  }
  
  @objc
  func setInitialViewpoint(_ node: NSNumber) {
    DispatchQueue.main.async {
      self.mappyMapViewContainer(node).setInitialViewpoint()
    }
  }
  
  private func mappyMapViewContainer(_ node: NSNumber) -> MappyMapViewContainer {
    self.bridge.uiManager.view(forReactTag: node) as! MappyMapViewContainer
  }
}
